File: project/2.03_インスタンスメソッドとインスタンス変数の概要/py/py_ex_2_03_1/src/ex_2_03_1.py
'''
Created on 2023/03/19

@author: Keroichi.T
'''

def main():
    s:str = "\"Romeo, where art thou?\""    
    
    # 文字列長から求める方法
    # 文字列要素をスライス([start:last]と指定)できる
    # startを省略すると0、lastを省略するとlen()とみなす
    print(s[len(s)-10:])

    
    # pythonの場合、startをマイナスにすることで末尾にまわりこむ
    # ただし要素数以上の数を指定しても先頭にはまわりこまない
    print(s[-10:])


if __name__ == '__main__':
    main()

File: project/2.03_インスタンスメソッドとインスタンス変数の概要/java/java_ex_2_03_1/src/java_ex_2_03_1/Ex_2_03_1.java
package java_ex_2_03_1;

public class Ex_2_03_1 {

	public static void main(String[] args) {
		String s = "\"Romeo, where art thou?\"";
		
		System.out.println(s.substring(s.length()-10));

		// ""で囲むことで、Stringオブジェクトは生成されている。
		System.out.println("\"Romeo, where art thou?\"".substring(0,10));
		
		
	}

}
